internal/bridge/crunchybridgecluster/crunchybridgecluster_controller.go

```diff
@@ -466,24 +511,24 @@ func (r *CrunchyBridgeClusterReconciler) reconcileBridgeConnectionSecret(
 	if err != nil {
 		meta.SetStatusCondition(&crunchybridgecluster.Status.Conditions, metav1.Condition{
 			ObservedGeneration: crunchybridgecluster.GetGeneration(),
-			Type:               v1beta1.ConditionCreating,
-			Status:             metav1.ConditionFalse,
+			Type:               v1beta1.ConditionReady,
+			Status:             metav1.ConditionUnknown,
 			Reason:             "SecretInvalid",
 			Message: fmt.Sprintf(
-				"Cannot create with bad secret: %v", "TODO(crunchybridgecluster)"),
+				"The secret is invalid: %v", err),
+		})
+		meta.SetStatusCondition(&crunchybridgecluster.Status.Conditions, metav1.Condition{
+			Type:               v1beta1.ConditionUpgrading,
+			Status:             metav1.ConditionUnknown,
+			ObservedGeneration: crunchybridgecluster.GetGeneration(),
+			LastTransitionTime: metav1.Time{},
+			Reason:             "SecretInvalid",
+			Message:            "The condition of the upgrade(s) is unknown because the secret is invalid",
```

Comment:
Let's put the error in the message:
```suggestion
			Message:            fmt.Sprintf("The condition of the upgrade(s) is unknown because the secret is invalid: %v", err),
```

Reply:
fixed